### PR TITLE
CI updates

### PR DESCRIPTION
**Pull request by ashnazg**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,24 +18,28 @@ install:
...
     - stage: lint
       php: 7.1
       before_script:
-        - composer create-project symplify/easy-coding-standard temp/ecs
+        - wget https://github.com/phpstan/phpstan/releases/download/0.9.1/phpstan.phar
       script:
-        - temp/ecs/bin/ecs check src tests
-        - vendor/bin/phpstan analyse src --level max --configuration phpstan.neon
+        - php phpstan.phar analyse src --level max --configuration phpstan.neon
+        - composer create-project symplify/easy-coding-standard temp/ecs && temp/ecs/bin/ecs check src tests
```
Comment (ashnazg) [inline .travis.yml]: I switched the order of these two, because the ecs install takes so much longer.

**Comment by ashnazg** [comment]
This one is ready now.  I'll add strict types on a follow-up PR.

**Comment by ashnazg** [comment]
Ok, *now* I reckon I have it done :)

**Comment by ashnazg** [comment]
I think this is ready now.

**Comment by jaapio** [comment]
@ashnazg I'm confused by these changes, You are installing phpunit in 3 ways. I don't quite get why. This this because you mixed up things trying to fix other projects?

**Comment by ashnazg** [comment]
@jaapio the travis builds were failing to install phpunit via composer.lock, every time on every PHP version.  I could never exactly resolve why, though I'm convinced it had to do with the reflection-docblock and reflection-common dependencies inside phpunit itself.  I'm fairly sure that's the same reason why the phpunit.phar would not generate code coverage for reflection-common, because its executed lines were being attached to the reflection-common inside phpunit.phar.

So, I ended up with this arrangement:
- phpunit remains in require-dev, and thus in composer.lock, for normal developer and IDE use
- since travis doesn't like what's in composer.lock, I have travis doing `composer install --no-dev` just to get reflection-common safely installed.  I then do the phpunit composer --dev install separately, and I guess composer is then able to derive a working dependency tree and allow the installation
- I know I had tried using the phpunit.phar as installed by phive.phar, but again, code coverage generation doesn't work, presumably due to the reflection-common dep inside phpunit.phar

**Comment by jaapio** [comment]
Thanks for this public explanation, this will help our contrubitors to understand what's going one. 

Read for merge.

**Pull request merged**